File: web/src/model/victron_controller/dashboard/from_0_2_0_core-state.ts
// @ts-nocheck
import {CoreState as v0_2_0_CoreState} from './v0_2_0/CoreState'
import {CoreState as dashboard_CoreState} from './CoreState'

export function convert__core_state__from__0_2_0(from: v0_2_0_CoreState): dashboard_CoreState {
    return new dashboard_CoreState (
        from.id,
        from.depends_on,
        from.last_run_outcome,
        from.last_payload,
        JSON.parse(JSON.stringify(from.last_inputs)),
        JSON.parse(JSON.stringify(from.last_outputs))
    )
}

File: crates/dashboard-model/src/victron_controller/dashboard/from_0_2_0_core_state.rs


pub fn convert__core_state__from__0_2_0(from: &crate::victron_controller::dashboard::v0_2_0::core_state::CoreState) -> crate::victron_controller::dashboard::core_state::CoreState {
    crate::victron_controller::dashboard::core_state::CoreState {
        id: from.id.clone(),
        depends_on: from.depends_on.clone(),
        last_run_outcome: from.last_run_outcome.clone(),
        last_payload: from.last_payload.clone(),
        last_inputs: serde_json::from_value(serde_json::to_value(&from.last_inputs).unwrap()).unwrap(),
        last_outputs: serde_json::from_value(serde_json::to_value(&from.last_outputs).unwrap()).unwrap(),
    }
}